src/sentry/runner/commands/upgrade.py

```diff
@@ -137,6 +138,11 @@ def _upgrade(interactive, traceback, verbosity, repair, with_nodestore):
 
         call_command("sentry.runner.commands.repair.repair")
 
+    # Used in getsentry.
+    cli_upgrade.send_robust(
```

Comment:
Does it make sense to call this `post_cli_upgrade`?

Reply:
Perhaps. I'm reconsidering this approach based on the feedback in the sibiling pull request.

I'm mulling over this solution instead:

```diff
diff --git a/src/sentry/runner/commands/upgrade.py b/src/sentry/runner/commands/upgrade.py
index 2006faee7c..964bb5cc93 100644
--- a/src/sentry/runner/commands/upgrade.py
+++ b/src/sentry/runner/commands/upgrade.py
@@ -3,7 +3,6 @@ from django.conf import settings
 from django.db import ProgrammingError, connections
 
 from sentry.runner.decorators import configuration
-from sentry.signals import cli_upgrade
 
 # List of migrations which we'll fake if we're coming from South
 DJANGO_MIGRATIONS = (
@@ -118,15 +117,22 @@ def _upgrade(interactive, traceback, verbosity, repair, with_nodestore):
     # migrate legacy south history into new django migrations automatically
     _migrate_from_south(verbosity)
 
-    dj_call_command(
-        "migrate",
-        interactive=interactive,
-        traceback=traceback,
-        verbosity=verbosity,
-        migrate=True,
-        merge=True,
-        ignore_ghost_migrations=True,
-    )
+    for db_conn in settings.DATABASES.keys():
+        # Always run migrations for the default connection.
+        # Also run migrations on connections that have migrations explicitly enabled.
+        # This is used for sentry.io as our production database runs on multiple hosts.
+        if db_conn == "default" or settings.DATABASES[db_conn].get("RUN_MIGRATIONS", False):
+            click.echo(f"Running migrations for {db_conn}")
+            dj_call_command(
+                "migrate",
+                database=db_conn,
+                interactive=interactive,
+                traceback=traceback,
+                verbosity=verbosity,
+                migrate=True,
+                merge=True,
+                ignore_ghost_migrations=True,
+            )
 
     if with_nodestore:
         from sentry import nodestore
@@ -138,11 +144,6 @@ def _upgrade(interactive, traceback, verbosity, repair, with_nodestore):
 
         call_command("sentry.runner.commands.repair.repair")
 
-    # Used in getsentry.
-    cli_upgrade.send_robust(
-        sender="cli", traceback=traceback, interactive=interactive, verbosity=verbosity
-    )
-
 
 @click.command()
 @click.option("--verbosity", "-v", default=1, help="Verbosity level.")
diff --git a/src/sentry/signals.py b/src/sentry/signals.py
index b447bb6cb3..a7c5724631 100644
--- a/src/sentry/signals.py
+++ b/src/sentry/signals.py
@@ -114,8 +114,6 @@ issue_deleted = BetterSignal(providing_args=["group", "user", "delete_type"])
 
 monitor_failed = BetterSignal(providing_args=["monitor"])
 
-cli_upgrade = BetterSignal(providing_args=["interactive", "traceback", "verbosity"])
-
 # experiments
 join_request_created = BetterSignal(providing_args=["member"])
 join_request_link_viewed = BetterSignal(providing_args=["organization"])

```

Reply:
I think this works well, and has the nice side benefit that OSS people could make use of this at some point.